tests/js/plugin/texthighlight-test.coffee

```diff
@@ -0,0 +1,78 @@
+assert = chai.assert
+sinon.assert.expose(assert, prefix: '')
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+describe 'Annotator.Plugin.TextHighlight', ->
+  sandbox = null
+  jqElement = null
+  scrollTarget = null
+
+  createTestHighlight = ->
+    anchor =
+      id: "test anchor"
+      annotation: "test annotation"
+      anchoring:
+        id: "test anchoring manager"
+        annotator:
+          id: "test annotator"
+          element:
+            delegate: sinon.spy()
+
+    new Annotator.TextHighlight anchor, "test page", "test range"
+
+  beforeEach ->
+    sandbox = sinon.sandbox.create()
+    sandbox.stub Annotator.TextHighlight, 'highlightRange',
+      (normedRange, cssClass) -> "test highlight span"
+
+    sandbox.stub Annotator.$.fn, "init", (selector, context) ->
+      jqElement =
+        selector: selector
+        data: sinon.spy()
+        scrollintoview: sinon.spy (options) ->
+          scrollTarget = this.selector
+          options?.complete?()
+
+  afterEach ->
+    sandbox.restore()
+    scrollTarget = null
+
+  describe "constructor", ->
+    it 'wraps a highlight span around the given range', ->
+      hl = createTestHighlight()
+      assert.calledWith Annotator.TextHighlight.highlightRange, "test range"
+
+    it 'stores the created highlight spans in _highlights', ->
```

Comment:
> If PDF.js doesn't create elements I'm skeptical of the claim that it's due to performance concerns.

It's not just about rendering performance; they are other reasons, too. For example, PDF.js supports partial loading of PDF files from the server, which means that the data is not even available in the browser, until it's specifically requested.

All in all, they (and many other viewer applications like them) just don't subscribe to the notion that the browser's `document` object is supposed to represent the actual underlying document somehow. They just use the web as a platform to run a viewer application.

In theory, we could try to salvage this situation by trying to conjure up new semantics for existing HTML elements, but I think the ultimate solution should be to define an API that acknowledges this distinction (ie. that the thing in or around the DOM is not the same as the document), and provides access to the underlying document in a way that can be applied to all these situations.

Anyway, I am willing to explore all future directions - but can we move on with these regression tests now?